spec/models/post_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject(:author) { User.create(name: 'Essie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Software Developer from Ethiopia.') }
+  subject(:post) { Post.create(title: 'hello world', text: 'Hello world paragraph', author_id: author.id) }
+
+  it 'should be a valid post' do
+    expect(post).to be_valid
+  end
+
+  it 'should be invalid post' do
+    post.title = nil
+    expect(post).to_not be_valid
+  end
+
+  it 'should have post counter to 1' do
+    expect(post.author.posts.size).to be(1)
+  end
+end
```

Comment:
Hi, thank you for submitting your code for review. I have checked your code and here are some feedbacks:

- Your code is well-structured and easy to read. You have used descriptive variable names and consistent indentation. Good job! 👏
- Your code follows the RSpec syntax and conventions. You have used `subject` to define the objects under test and `expect` to assert the expected outcomes. Well done! 👍
- Your code covers the basic validations for the Post model, such as presence and length of title. However, you have missed some validations that are required by the project specifications, such as:
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- You can add these validations to your Post model using the `numericality` option, for example:
- `validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }`
- `validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }`
- You should also add corresponding tests to your RSpec file to check these validations, for example:
- `it 'should be invalid if comments_counter is negative' do`
- `post.comments_counter = -1`
 - `expect(post).to_not be_valid`
 - `end`
- `it 'should be invalid if likes_counter is negative' do`
- `post.likes_counter = -1`
- `expect(post).to_not be_valid`
- `end`
- These validations and tests will ensure that your Post model meets the project requirements and prevents invalid data from being saved to the database.

I hope this feedback helps you improve your code. Please let me know if you have any questions or need further assistance. Happy coding! 😊